src/main/java/org/stellar/base/Transaction.java

```diff
@@ -1,14 +1,151 @@
 package org.stellar.base;
 
-import org.stellar.base.xdr.Operation;
+import org.apache.commons.codec.binary.Base64;
+import org.stellar.base.xdr.EnvelopeType;
+import org.stellar.base.xdr.DecoratedSignature;
+import org.stellar.base.xdr.PublicKey;
+import org.stellar.base.xdr.Signature;
+import org.stellar.base.xdr.SignatureHint;
+import org.stellar.base.xdr.XdrDataOutputStream;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 public class Transaction {
+  private final int BASE_FEE = 100;
+
+  private final int mFee;
+  private final Account mSourceAccount;
+  private final long mSequenceNumber;
+  private final Operation[] mOperations;
+  private final org.stellar.base.xdr.Memo mMemo;
+  private List<DecoratedSignature> mSignatures;
+
+  Transaction(Account sourceAccount, long sequenceNumber, Operation[] operations, org.stellar.base.xdr.Memo memo) {
+    if (operations.length == 0) {
+      throw new RuntimeException("At least one operation required.");
+    }
+
+    mSourceAccount = sourceAccount;
+    mSequenceNumber = sequenceNumber;
+    mOperations = operations;
+    mFee = operations.length * BASE_FEE;
+    mSignatures = new ArrayList<DecoratedSignature>();
+    mMemo = memo != null ? memo : Memo.none();
+  }
+
+  public void sign(StellarKeypair signer) throws IOException {
```

Comment:
IMO, the body of this method should be simplified down to:
```java
byte[] txHash = this.hash();
mSignatures.add(signer.signDecorated(txHash));
```

IMO, the responsibility of creating a DecoratedSignature should be with the keypair, just as it is with creating a raw signature.